src/indicators/flooding/glofas/utils.py

```diff
@@ -257,6 +259,68 @@ def get_glofas_activations(da_glofas, thresh, ndays):
     return df_glofas_act
 
 
+def get_detection_stats(df_glofas, df_impact, buffer_before, buffer_after):
+    TP = 0
+    FP = 0
+    tot_events = len(df_impact.index)
+    df_impact_copy = df_impact.copy()
+
+    # Add buffer around the flood event dates to account for some uncertainty if desired
+    df_impact_copy["start_date_buffer"] = pd.to_datetime(
+        df_impact_copy["start_date"]
+    ) - timedelta(days=buffer_before)
+    df_impact_copy["end_date_buffer"] = pd.to_datetime(
+        df_impact_copy["end_date"]
+    ) + timedelta(days=buffer_after)
+
+    for index, row in df_glofas.iterrows():
+        TP_ = False
+        act_dates = np.array(pd.date_range(row["start_date"], row["end_date"]))
+
+        for index, row in df_impact_copy.iterrows():
+            event_dates = np.array(
+                pd.date_range(row["start_date_buffer"], row["end_date_buffer"])
+            )
+
+            if set(act_dates) & set(event_dates):
+                TP += 1
+                TP_ = True
+                df_impact_copy = df_impact_copy.drop([index,])
```

Comment:
I'm worried that this might cause you to overcount false positives. Let's say you have two GloFAS events that are nearly on the same day. Then the first one will be counted as a TP, and the impact event dropped. The second GloFAS event will then be counted as an FP because the nearby GloFAS event will no longer be in df_impact_copy. 

To fix this what you could do is loop first over df_impact, and have a column in the dataframe to indicate whether it's been detected or not. Then for each GloFAS event you check if it overlaps (TP) or does not (FP). I hope that makes sense.

Reply:
Ah good point - thanks for catching this! It would be great if this reduced the false positive rate. I haven't had the chance to implement this fix today, but will prioritize for when I'm back.

Reply:
@turnerm have made a separate branch here to address this: https://github.com/OCHA-DAP/pa-anticipatory-action/pull/135